subsys/net/ip/connection.c

```diff
@@ -828,11 +828,12 @@ enum net_verdict net_conn_input(struct net_pkt *pkt,
 
 	if (IS_ENABLED(CONFIG_NET_IP) && (pkt_family == AF_INET || pkt_family == AF_INET6) &&
 	    !(is_mcast_pkt || is_bcast_pkt)) {
-		conn_send_icmp_error(pkt);
-
 		if (IS_ENABLED(CONFIG_NET_TCP) && proto == IPPROTO_TCP) {
+			net_tcp_reply_rst(pkt);
```

Comment:
In Linux one can use netfilter to prevent RST to be sent out, and in Zephyr that might be possible at some point (currently not possible). So I was thinking that should we have a Kconfig option to allow earlier behavior, so that RST would not be sent if option is set, by default the option could be off so RST would be sent out. WDYT?

Reply:
@jukkar I've added a Kconfig option to configure whether to send RST packet on closed port. If disabled, ICMP is sent instead as it used to.